build.gradle

```diff
@@ -150,10 +150,30 @@ idea {
     }
 }
 
+tasks.register('lineEndingConvCopy', CRLFConvertCopy) {
```

Comment:
Can you help me understand why we need this? Is Windows-style CRLF the standard for these types of files in a JAR?

Reply:
For the jar being reproducible, one must standardize on either Unix or Windows line endings for the text files inside, otherwise Unix and Windows machines would produce different jar files. I chose Windows because:

a) Market share of Windows is larger.
b) Unix line endings could cause problems when viewed with some Windows tools like Notepad, the other way around is less likely to be problematic.
c) Unix/Linux users generally know about line endings, Windows users often do not.

I do that conversion in _all_ of my projects; do not know who else does that. And now that I think about it, I need to convert the `MANIFEST.MF` file as well...

If you are uncomfortable with that conversion, I could remove it. Note that the original license text sources are left unchanged, otherwise there could be some Git glitches.

Reply:
> And now that I think about it, I need to convert the `MANIFEST.MF` file as well...

Manifest files are written with Manifest.write() which always use CRLF, no need to convert. What needs to be uniformized are the resource files, that could be switched to CRLF (or to Unix line endings) once they have been copied to `buildGradle/resources/main` during the build.

Do I proceed with the line ending conversion of the resources, or should I remove the conversion stuff completely?

Reply:
My first choice would be to not have the line ending conversion code in there, but ensure using .gitattributes or some other suitable mechanism that the line endings are consistent. 

I would also prefer to keep them Unix-style since the majority of the open source world lives in the Unix world (even on Windows). And this is, after all, a Mozilla project.

Is there a way that we can have this without the conversion code using .gitattributes?

Reply:
> Is there a way that we can have this without the conversion code using .gitattributes?

Yes, although a `git add --renormalize .` is recommended for people on Windows. I'll push it soon.

Reply:
The idea was to have the following `.gitattributes`:
```gitattributes
# Set default behavior, in case users don't have core.autocrlf set.
* text=auto eol=lf

.classpath text
.project text
*.html text eol=lf
*.htm text eol=lf
*.java text eol=lf
*.js text eol=lf
*.jstest text eol=lf
*.jsp text eol=lf
*.php text eol=lf
*.properties text eol=lf
*.sh text eol=lf
*.xml text eol=lf
*.md text eol=lf
*.bat text eol=crlf
*.gradle text eol=lf
*.pl text eol=lf
*.mk text eol=lf
*.tests text eol=lf
*.doctest text eol=lf
/man/*.1 text eol=lf
/gradlew text eol=lf

*.bmp binary
*.gif binary
*.jar binary
*.jpg binary
*.jpeg binary
*.png binary
```
Unfortunately, this breaks spotless. I'll push the commit without it, and then it can be addressed later (after 1.7.14 when spotless is applied to the remaining files).

By then, the following `.editorconfig` file could be added as well:
```ini
#
# https://editorconfig.org/
#
root = true

[*]
end_of_line = lf
charset = utf-8

[*.{java,js,gradle}]
indent_style = space
indent_size = 4

[*.bat]
end_of_line = crlf
charset = latin1
```